test/timeout.test.ts

```diff
@@ -0,0 +1,89 @@
+describe('timeout with abort controller', () => {
+    it('empty', async () => {
+        // empty test so jest won't fail in node < 15 due to no tests being executed
+    });
+
+    if (Number(process.version.match(/v(\d+)/)?.[1] ?? null) < 15) {
+        // skip tests as this package requires node 15+
+        return;
+    }
+
+    // we need to import via require after the above check to not fail on node < 15
+    // eslint-disable-next-line @typescript-eslint/no-var-requires,global-require
+    const { addTimeoutToPromise, tryCancel } = require('@apify/timeout');
+    // eslint-disable-next-line @typescript-eslint/no-var-requires,global-require
+    const { setTimeout } = require('timers/promises');
+
+    let position = 0;
+
+    async function handler() {
+        await setTimeout(30);
+        tryCancel();
+        position++;
+        await setTimeout(30);
+        tryCancel();
+        position++;
+        await setTimeout(30);
+        tryCancel();
+        position++;
+        await setTimeout(30);
+        tryCancel();
+        position++;
+        await setTimeout(30);
+        tryCancel();
+        position++;
+
+        return 123;
+    }
+
+    beforeEach(() => {
+        position = 0;
+    });
+
+    it('passes without timeouting', async () => {
+        const res = await addTimeoutToPromise(
+            () => handler(),
+            200,
+            'timed out',
+        );
```

Comment:
It might be worth using [fake timers](https://jestjs.io/docs/timer-mocks) to ensure the tests aren't flaky under load. But it might not be possible in this complex scenario.

Reply:
Lets deal with that when it starts happening, it feels like it will be safer to test it works with real timeouts properly.

Reply:
I agree with @B4nan. We're using fake timers in tests in Got and some internal functions need to be patched. I spent hours trying to get them to work and in the end gave up. We can always run hundreds of small, real timeouts instead.

Reply:
Well, we were not using fake timers in the SDK and we had A LOT of flaky tests 😄 But, ok, let's wait what happens 👍